File: backend-java/src/main/java/com/asmolabs/zanshin/repository/entities/VexDecision.java
package com.asmolabs.zanshin.repository.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "vex_decision")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VexDecision {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "vulnerability_id", nullable = false)
    private String vulnerabilityId;

    @Column(name = "package_name", nullable = false)
    private String packageName;

    private String purl;

    @Column(nullable = false)
    private String status;

    private String justification;

    private String response;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "repository_id")
    private ZanshinRepository repository;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: zanshin/models/vex_decision.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, ForeignKey, BigInteger
from sqlalchemy.orm import relationship
from zanshin.database import Base
from zanshin.models.safedatetime import SafeDateTime

class VexDecision(Base):
    __tablename__ = "vex_decision"

    id = Column(Integer, primary_key=True, index=True)
    vulnerability_id = Column(String(255), nullable=False)
    package_name = Column(String(255), nullable=False)
    purl = Column(String(255), nullable=True)
    status = Column(String(255), default="under_review", nullable=False)  # affected, not_affected, fixed, under_review
    justification = Column(String(255), nullable=True)
    response = Column(String(255), nullable=True)
    comment = Column(Text, nullable=True)
    created_at = Column(SafeDateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(SafeDateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    repository_id = Column(BigInteger, ForeignKey("repository.id"), nullable=True)
    repository = relationship("ZanshinRepository", back_populates="vex_decisions")
